languages/csharp/exercises/concept/numbers-floating-point/.docs/hints.md

```diff
@@ -14,7 +14,7 @@
 
 ### 3. Calculate the years before reaching the desired balance
 
-- To calculate the years, one has to loop over the balance. C# has several [looping constructs][docs.microsoft.com-loops].
+- To calculate the years, one has to keep looping until the desired balance is reached. C# has several [looping constructs][docs.microsoft.com-loops].
```

Comment:
I wonder if we want to make this more of "one option to do it". There are other options after all.

```suggestion
- To calculate the years, one can keep looping until the desired balance is reached. C# has several [looping constructs][docs.microsoft.com-loops].
```

Reply:
Excellent suggestion!

Reply:
On the other hand, if this is an introduction to basic-loops we might want to have something where there are less options and ask the student to solve it with a loop (instead of e.g. using a recursive call which might be possible as an alternative).

With this exercise I personally would say the logarithmic solution is the "best".

Reply:
We could have an analyzer that checks for logarithms and says "Logarithms are a great super-power to have used. It might also be worth you trying to solve this with a loop, so you've familiarised yourself with it in C#"

Reply:
(This is the sort of way I really want to use analyzers/representers in v3)

Reply:
Yep, me too!

Reply:
How about changing the task a bit by adding another parameter for yearly substraction/additions:

FloatingPointNumbers.YearsBeforeDesiredBalance(balance: 200.75m, targetBalance: 214.88m, yearlyAddition: -0.5m)

Then logarithms are off the table.

Reply:
Feel free to resolve again if that makes it too complex

Reply:
I think we should merge this, as it's a step forward. And then if C# decide to remove the ability for people to use logarithms (which sounds like a sensible suggestion), that can be a follow-up PR later.

I wonder if there's a value in having an IDEAS.md file by each exercise where things like that can be noted. We have a long time (relatively) between now and launch, and when tracks go through a refinement stage, it might be nice to have logs of these ideas for improvements there.

Reply:
I like the ideas of a scratchpad `ideas.md` files !